test/MaeBlockTest.cpp

```diff
@@ -1,7 +1,6 @@
 #include <iostream>
 #include <stdexcept>
 
-#define BOOST_TEST_DYN_LINK
```

Comment:
For some reason I thought that this was needed to actually run the tests. Are you sure that tests are actually being executed? Like, not that the file is being called by ctest, but that when it is called it is actually running tests?

When I remove this from a test that I wrote, it doesn't print any output or respond to expected failures, even when I manually added a throw and segv to one of the tests.

Reply:
If I understand it correctly, that macro is required to build the tests if you are going to link against the boost unit test framework dynamically. If it is in the file, it won't be able to link statically, as boost will try to provide a dynamically linked `main()` function while also trying to provide a static one.

I think this is correct, since it must not be provided to static tests, and in dynamically linked ones, it will be provided by the global `BOOST_ALL_DYN_LINK` that is set in the CMake file.

Reply:
I have confirmed that the tests run correctly (both when building static and shared) by forcing a test to fail.

Reply:
Thanks, man. Sorry for the paranoia.